packages/x-charts/src/internals/configInit.ts

```diff
@@ -0,0 +1,28 @@
+import { ChartSeriesType } from '../models/seriesType/config';
+
+let instance: undefined | Set<ChartSeriesType>;
+
+class CartesianSeriesTypes {
+  types: Set<ChartSeriesType> = new Set();
+
+  constructor() {
+    if (instance) {
+      throw new Error('You can only create one instance!');
+    }
+    instance = this.types;
+  }
+
+  addType(value: ChartSeriesType) {
+    this.types.add(value);
+  }
+
+  getTypes() {
+    return this.types;
+  }
+}
+
+export const cartesianSeriesTypes = new CartesianSeriesTypes();
+
+cartesianSeriesTypes.addType('bar');
+cartesianSeriesTypes.addType('line');
+cartesianSeriesTypes.addType('scatter');
```

Comment:
If we always want these in, we could just move them to the `new Set(['bar', 'line', 'scatter'])`

Reply:
That's what I did when copy/past your proposal, but for TS to work, we should do `new Set(['bar', 'line', 'scatter'] as const)`

The I realized using `cartesianSeriesTypes.addType` would
- please TS without `as`
- do the same thing
- bring consistency on how we define series type as cartesian in community and pro package (`cartesianSeriesTypes.addType('heatmap');`)